NASA_API.pyw

```diff
@@ -20,13 +21,18 @@ def main():
     # requesting data from hdurl
     image_data = requests.get(hdurl)
 
-    # creating path and name of image
-    current_dir = str(pathlib.Path(__file__).parent.resolve())
-    image_path = current_dir + '/APOD_images/%s_NASA_APOD.png' %date
-
-    # saving data from hdurl into a png file
-    with open(image_path, 'wb') as f:
-        f.write(image_data.content)
+    # creating path and name of image, and saving data from hdurl into a png file 
+    try:
+        current_dir = str(pathlib.Path(__file__).parent.resolve())
+        image_path = current_dir + '/APOD_images/%s_NASA_APOD.png' %date
```

Comment:
In software development there is a term 'DRY' which means 'Do not repeat yourself', it applies for most things, lines of code, function, classes, etc.

The idea is to have the a single source of truth, which  in case you need to make changes, you only do them in 1 place.

In this case, lines 27-29 is repeated twice, can you refactor it. (Refactor) means to re-write the code without changing its behaviour. It should work the same, just written differently.